### PR TITLE
feat: add source tag materialized view

### PR DESCRIPTION
**Pull request by rebelchris**
Added a materialized view for source-tag-count 
Will modify use-cases in a separate PR to keep things easier to review.

Local example:
Screenshot 2024-04-12 at 16 13 52

### CODE REVIEW

File: src/entity/SourceTagView.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      })
+      .innerJoin(
+        PostKeyword,
+        'pk',
+        'pk.postId = p.id AND pk.status = :status',
+        { status: 'allow' },
+      )
+      .andWhere({ active: true, private: false })
+      .groupBy(`sourceId, tag`)
+      .orderBy('count', 'DESC'),
```
Comment (idoshamun) [inline src/entity/SourceTagView.ts:27]: We don't really need order by but not blocking

File: src/entity/SourceTagView.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+})
+export class SourceTagView {
```
Comment (idoshamun) [inline src/entity/SourceTagView.ts:29]: We need index for source and index for tag

**Review by idoshamun — APPROVED** [review]
Non blocking comments

File: src/entity/SourceTagView.ts (hunk 1 above)
Comment (rebelchris) [reply src/entity/SourceTagView.ts:27]: Thinking was it was cheaper to do here then in other queries using this?

Comment (idoshamun) [reply src/entity/SourceTagView.ts:27]: Doing it here doesn't necessarily mean downstream queries will be ordered

Comment (rebelchris) [reply src/entity/SourceTagView.ts:27]: Happy to remove it here 👍 will do with the indexes in one go

**Review by omBratteng — APPROVED** [review]

**Review by capJavert — APPROVED** [review]

**Pull request merged**